xdsl/dialects/memref.py

```diff
@@ -274,14 +274,14 @@ class Alloca(IRDLOperation):
     memref: OpResult = result_def(MemRefType[Attribute])
 
     # TODO how to constraint the IntegerAttr type?
-    alignment: AnyIntegerAttr = attr_def(AnyIntegerAttr)
+    alignment: AnyIntegerAttr | None = opt_attr_def(AnyIntegerAttr)
 
     irdl_options = [AttrSizedOperandSegments()]
 
     @staticmethod
     def get(
         return_type: Attribute,
-        alignment: int,
+        alignment: int | None = None,
```

Comment:
```suggestion
        alignment: int | IntegerAttr | None = None,
```
Should probably accept alignment that's already an attribute also, in case we lower from somewhere else.

Reply:
Wouldn't it make more sense to make another PR for that since it's not the only operation that could be improved in the same way (e.g: memref.alloc) ?

Reply:
This is the code that you're changing so I think it's worth doing as you go, it's an improvement as far as I can tell. We very rarely have the time to clean code in a dedicated way, so adding small improvements like this seems like a good idea to me. It's hard to say where to draw the line exactly, more of a feel thing, so it's not a blocker.

Reply:
But actually this particular change would also clean up the code below, where you would avoind the if expression in the init call